[FIX] default_warehouse_from_sale_team: access error when chained rule i#17897

luisg123v opened the pull request:
When a stock movement triggers an inventory rule that involves another
warehouse on which the current user has no access, an access error is
triggered. That's because, thanks to this module, users can only see
warehouses allowed in their sales teams.

To solve the above, access is granted to all warehouses when computing
inventory rules to apply.

### Dummy MR:

- [vauxoo/typ!936](https://git.vauxoo.com/vauxoo/typ/-/merge_requests/936)

luisg123v commented:
@desdelinux and (@keylor2906 or @hugho-ad),

Could you review, please?

FYI reusing a dummy again, but this change is applied there.

Regards,

hugho-ad commented:
why in v11 this error was not happen?

luisg123v commented:
@hugho-ad,

Because in v11 there was [this](https://github.com/Vauxoo/addons-vauxoo/blob/2719ebed2246896c17228cebc16fce9808459f46/default_warehouse_from_sale_team/models/sales_team.py#L88).

desdelinux reviewed (approved):
LGTM

It also solves the issue of manual transfers between warehouses

@hugho-ad @luisg123v please merged

hugho-ad commented on default_warehouse_from_sale_team/models/procurement_group.py:
```diff
@@ -0,0 +1,23 @@
+            and warehouse_id.check_access_rights("read", raise_exception=False)
+            and self.env["ir.rule"]._get_failing(warehouse_id, mode="read") == rule_warehouse
+        ):
+            warehouse_id = warehouse_id.sudo()
```
why not by default always use sudo as the old approach?

desdelinux replied:
@luisg123v @hugho-ad I think that using sudo for everything we could also solve this case: https://www.vauxoo.com/web#id=17904&action=1918&model=helpdesk.ticket&view_type=form&cids=1&menu_id=1431

luisg123v replied:
@hugho-ad I would like to avoid that approach, as we would be forcin sudo in all situations, independently of whether is required.

FYI the other issue commented by @desdelinux is not solved by adding sudo, because it's an access error on a different model (picking).

luisg123v committed:
[FIX] default_warehouse_from_sale_team: access error when chained rule

When a stock movement triggers an inventory rule that involves another
warehouse on which the current user has no access, an access error is
triggered. That's because, thanks to this module, users can only see
warehouses allowed in their sales teams.

To solve the above, access is granted to all warehouses when computing
inventory rules to apply.
